edc-tests/edc-dataplane/cloud-transfer-tests/src/test/java/org/eclipse/tractusx/edc/dataplane/transfer/test/AzureToAzureTest.java

```diff
@@ -0,0 +1,244 @@
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.eclipse.edc.junit.testfixtures.TestUtils.getFreePort;
+import static org.eclipse.tractusx.edc.dataplane.transfer.test.TestFunctions.createSparseFile;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.Mockito.verify;
+import static org.testcontainers.shaded.org.awaitility.Awaitility.await;
+
+/**
+ * This test is intended to verify transfers within the same cloud provider, i.e. AzureBlob-to-AzureBlob.
+ * It spins up a fully-fledged dataplane and issues the DataFlowRequest via the data plane's Control API
+ */
+@SuppressWarnings("resource")
+@Testcontainers
+@AzureStorageIntegrationTest
+public class AzureToAzureTest {
+    public static final String PROVIDER_ACCOUNT_NAME = "provider";
+    public static final String PROVIDER_ACCOUNT_KEY = "providerkey";
+    public static final String TESTFILE_NAME = "testfile.json";
+    public static final String CONSUMER_ACCOUNT_NAME = "consumer";
+    public static final String CONSUMER_ACCOUNT_KEY = "consumerkey";
+    // alias under which the provider key is stored in the vault. must end with -key1
+    public static final String PROVIDER_KEY_ALIAS = "providerkey-key1";
+    // alias under which the consumer key is stored in the vault. must end with -key1
+    public static final String CONSUMER_KEY_ALIAS = "consumerkey-key1`";
+    private static final int PROVIDER_CONTROL_PORT = getFreePort();
+
+    private static final int AZURITE_HOST_PORT = getFreePort();
+    // launches the data plane
+    @RegisterExtension
+    protected static final ParticipantRuntime DATAPLANE_RUNTIME = new ParticipantRuntime(
+            ":edc-tests:runtime:dataplane-cloud",
+            "AzureBlob-Dataplane",
+            RuntimeConfig.Azure.createDataplane("/control", PROVIDER_CONTROL_PORT, AZURITE_HOST_PORT)
+    );
+    private static final String AZURITE = "mcr.microsoft.com/azure-storage/azurite";
+    private static final String COMPLETION_MARKER = ".complete";
+    private static final String PROVIDER_CONTAINER_NAME = "src-container";
+    private static final String CONSUMER_CONTAINER_NAME = "dest-container";
+    /**
+     * Currently we have to use one container to host both consumer and provider accounts, because we cannot handle
+     * two different endpoint templates for provider and consumer. Endpoint templates are configured globally.
+     * Also, the host-port must be fixed/deterministic, as the {@code PROVIDER_RUNTIME} needs to know it in advance
+     */
+    @Container
+    private final FixedHostPortGenericContainer<?> azuriteContainer = new FixedHostPortGenericContainer<>(AZURITE)
+            .withFixedExposedPort(AZURITE_HOST_PORT, 10000)
+            .withEnv("AZURITE_ACCOUNTS", PROVIDER_ACCOUNT_NAME + ":" + PROVIDER_ACCOUNT_KEY + ";" + CONSUMER_ACCOUNT_NAME + ":" + CONSUMER_ACCOUNT_KEY); //provider and consumer account in the same instance
```

Comment:
Comment can be moved into the bigger one above to avoid overlong lines.
Almost missed this comment bacause of that.